src/Module/Contact/Follow.php

```diff
@@ -0,0 +1,241 @@
+		}
+
+		$contact = Contact::getByURL($url, true);
+
+		// Possibly it is a mail contact
+		if (empty($contact)) {
+			$contact = Probe::uri($url, Protocol::MAIL, $uid);
+		}
+
+		if (empty($contact) || ($contact['network'] == Protocol::PHANTOM)) {
+			// Possibly it is a remote item and not an account
+			$this->followRemoteItem($url);
+
+			$this->sysMessages->addNotice($this->t('The network type couldn\'t be detected. Contact can\'t be added.'));
+			$submit  = '';
+			$contact = ['url' => $url, 'network' => Protocol::PHANTOM, 'name' => $url, 'keywords' => ''];
+		}
+
+		$protocol = Contact::getProtocol($contact['url'], $contact['network']);
+
+		if (($protocol == Protocol::DIASPORA) && !$this->config->get('system', 'diaspora_enabled')) {
+			$this->sysMessages->addNotice($this->t('Diaspora support isn\'t enabled. Contact can\'t be added.'));
+			$submit = '';
+		}
+
+		if (($protocol == Protocol::OSTATUS) && $this->config->get('system', 'ostatus_disabled')) {
+			$this->sysMessages->addNotice($this->t("OStatus support is disabled. Contact can't be added."));
+			$submit = '';
+		}
+
+		if ($protocol == Protocol::MAIL) {
+			$contact['url'] = $contact['addr'];
+		}
+
+		if (!empty($request['auto'])) {
+			$this->process($contact['url']);
+		}
+
+		$request = $this->baseUrl . '/follow';
+		$tpl     = Renderer::getMarkupTemplate('auto_request.tpl');
+
+		$owner = User::getOwnerDataById($uid);
+		if (empty($owner)) {
+			$this->sysMessages->addNotice($this->t('Permission denied.'));
+			$this->baseUrl->redirect($returnPath);
+		}
+
+		$myaddr = $owner['url'];
+
+		$output = Renderer::replaceMacros($tpl, [
+			'$header'         => $this->t('Connect/Follow'),
+			'$pls_answer'     => $this->t('Please answer the following:'),
+			'$your_address'   => $this->t('Your Identity Address:'),
+			'$url_label'      => $this->t('Profile URL'),
+			'$keywords_label' => $this->t('Tags:'),
+			'$submit'         => $submit,
+			'$cancel'         => $this->t('Cancel'),
+
+			'$request'  => $request,
+			'$name'     => $contact['name'],
+			'$url'      => $contact['url'],
+			'$zrl'      => Profile::zrl($contact['url']),
+			'$myaddr'   => $myaddr,
+			'$keywords' => $contact['keywords'],
+
+			'$does_know_you' => ['knowyou', $this->t('%s knows you', $contact['name'])],
+			'$addnote_field' => ['dfrn-request-message', $this->t('Add a personal note:')],
+		]);
+
+		$this['aside'] = '';
+
+		if (!in_array($protocol, [Protocol::PHANTOM, Protocol::MAIL])) {
+			$this['aside'] = VCard::getHTML($contact);
+
+			$output .= Renderer::replaceMacros(Renderer::getMarkupTemplate('section_title.tpl'),
+				['$title' => $this->t('Status Messages and Posts')]
+			);
+
+			// Show last public posts
+			$output .= Contact::getPostsFromUrl($contact['url']);
+		}
+
+		return $output;
+	}
+
+	protected function process(string $url)
+	{
+		$returnPath = 'follow?rul=' . urlencode($url);
+
+		$result = Contact::createFromProbeForUser($this->app->getLoggedInUserId(), $url);
```

Comment:
Are we sure we don't need `UserSession->getLocalUserId()` instead?

Reply:
What about the Hook call inside the method for the event `follow`, this seems related to this process

Reply:
I meant in replacement of `$this->app->getLoggedInUserId()`
```suggestion
		$result = Contact::createFromProbeForUser($this->userSession-> getLocalUserId(), $url);
```

Reply:
Ah oh sry, I thought you mean the whole Contact method :D